services/workflows-service/src/document/helpers/add-requested-document-to-entity-documents.ts

```diff
@@ -0,0 +1,122 @@
+import { AnyRecord } from '@ballerine/common';
+import { Document, UiDefinition } from '@prisma/client';
+import get from 'lodash/get';
+import set from 'lodash/set';
+
+export interface IDocumentTemplate {
+  id: string;
+  category: string;
+  type: string;
+  issuer: {
+    country: string;
+  };
+  version: number;
+  issuingVersion: number;
+  properties: AnyRecord;
+  pages: AnyRecord[];
+  status?: Document['status'];
+  decision?: Document['decision'];
+  // Id of an document without file.
+  _id: string;
+}
+
+interface IUIDefinitionPage {
+  elements: IUIDefinitionElement[];
+}
+
+export interface IUIDefinitionElement {
+  element: string;
+  params?: {
+    template?: {
+      id: string;
+      type: string;
+      category: string;
+    };
+  };
+  children?: IUIDefinitionElement[];
+}
+
+const findDocumentDefinitionByTypeAndCategory = (
+  type: string,
+  category: string,
+  uiDefinition: UiDefinition,
+): IUIDefinitionElement | undefined => {
+  let result: IUIDefinitionElement | undefined = undefined;
+  const pages = (uiDefinition.uiSchema as unknown as { elements: IUIDefinitionPage[] }).elements;
+
+  const run = (elements: IUIDefinitionElement[]) => {
+    for (const element of elements) {
+      if (
+        element.element === 'documentfield' &&
+        element?.params?.template?.type === type &&
+        element?.params?.template?.category === category
+      ) {
```

Comment:
This logic needs to be the same as the one in getDocumentId and in the backoffice. Where we also take into consideration fields like version and country.